src/directives/admonition.ts

```diff
@@ -1,31 +1,44 @@
-import { toHTML } from '../utils';
-import { DirectiveConstructor } from './types';
+import { Directive } from './types';
 
 const admonitionTitles = {
   attention: 'Attention', caution: 'Caution', danger: 'Danger', error: 'Error', important: 'Important', hint: 'Hint', note: 'Note', seealso: 'See Also', tip: 'Tip', warning: 'Warning',
 };
 const DEFAULT_ADMONITION_CLASS = 'note';
 type AdmonitionTypes = keyof typeof admonitionTitles | 'admonition';
 
-const createAdmonition = (kind: AdmonitionTypes): DirectiveConstructor => {
+export type Args = {
+  title: string;
+};
+
+export type Opts = {
+  class: AdmonitionTypes;
+};
+
+const createAdmonition = (kind: AdmonitionTypes): Directive<Args, Opts> => {
   const className = kind === 'admonition' ? DEFAULT_ADMONITION_CLASS : kind;
   return {
     token: kind,
     getArguments: (info) => {
       const content = kind === 'admonition' ? '' : info;
       const title = kind === 'admonition' ? info : admonitionTitles[kind];
-      const attrs = { title };
-      return { attrs, content };
+      const args = { title };
+      return { args, content };
+    },
+    getOptions: (data) => {
+      const { class: overrideClass, ...rest } = data;
+      if (Object.keys(rest).length > 0) {
+        console.warn('Unknown admonition options');
+      }
+      return { class: overrideClass as AdmonitionTypes };
     },
-    getOptions: (data) => data,
-    renderer: (tokens, idx) => {
-      const token = tokens[idx];
-      const title = token.attrGet('title') ?? '';
-      return toHTML([
-        'aside', { class: ['callout', className] },
+    renderer: (args, opts) => {
+      const { title } = args;
+      const { class: overrideClass } = opts;
```

Comment:
Instead of having to figure out where on markdown-it the args and opts are help, they are now the first input into this function.